duckduckgo/iOS
# Manage Expired Subscription

Task/Issue URL: https://app.asana.com/0/1204099484721401/1207112724044306/f

**Description**:
- Allows repurchasing an expired subscription, whether it was purchased on Apple or via Stripe or Google
- Properly caches Settings Subscription-State and uses it by default (Until the backend says otherwise)



**Test Repurchase**:
1. Change FE URL to use abrown.duckduckgo.com
2. Get a Stripe subscription
3. Restore on iOS using this version
4. Expire Stripe subscription
5. Open Settings
6. Observe the subscription is displayed as expired
7. Tap View Plans
8. Purchase a subscription with the app
9. Observe that the subscription is active after purchase and that the email address is still there.

**Test caching**
1. After you have an active subscription
2. Disable Network Access
3. Force close app
4. Open App with network off
5. Observe Subscription is active in settings
6. Tap ITP
7. Observe an error appears and you're taken back to settings
8. Go to Subscription Settings
9. Remove from Device
10. Observe you're signed out
11. Tap Get Privacy Pro
12. Observe you get an error message that takes you back to settings
13. Tap I have a subscription
14. Tap "Enter Email"
15. Observe you get an error message that takes you back
16. Tap "Restore Purchase"
17.  Observe you Get an error message that takes you back to settings
